docs/docs/nlu-training-data.mdx

```diff
@@ -3,92 +3,166 @@ id: nlu-training-data
+Synonym mapping only happens **after** entities have been extracted.
+That means that your training examples should include the synonym examples
+(`credit card account` and `credit account`) so that the model will learn to
+recognize these as entities and replace them with `credit`.
+:::
 
-2. They can be used to directly extract entities from a user messages when using the [RegexEntityExtractor](./components.mdx#regexentityextractor) component in the pipeline.
+See the [training data format](./training-data-format.mdx) for details on how to include synonyms in your training data.
 
-For example, if your entity has a deterministic structure (like a zipcode or an email address), you can use a regular
-expression to ease detection of that entity (using the [RegexFeaturizer](./components.mdx#regexfeaturizer)) or to directly extract the entities from
-the user message (using the [RegexEntityExtractor](./components.mdx#regexentityextractor)).
+## Regular Expressions
 
+You can use regular expressions to improve intent classification and
+entity extraction in combination with the [`RegexFeaturizer`](components.mdx#regexfeaturizer) and [`RegexEntityExtractor`](components.mdx#regexentityextractor) components in the pipeline.
 
-In case you are using the regular expressions for the [RegexFeaturizer](./components.mdx#regexfeaturizer) the name of the regular expression does
-not matter. If does not define the entity nor the intent, it is just a human readable description for you to remember
-what this regex is used for and is the title of the corresponding pattern feature.
-If you want to use the [RegexFeaturizer](./components.mdx#regexfeaturizer) you can also use the regex features to improve the intent
-classification performance, for example, by defining a greet clause:
 
+### Regular Expressions for Intent Classification
 
-If you are using the regular expressions to directly extract entities using the [RegexEntityExtractor](./components.mdx#regexentityextractor),
-the name of the regular expression should match the name of the entity you want to extract.
+You can use regular expressions to improve intent classification by including the `RegexFeaturizer` component in your pipeline. When using the `RegexFeaturizer`, a regex does not act as a rule for classifying an intent. It only provides a feature that the intent classifier will use
+to learn patterns for intent classification.
+Currently, all intent classifiers make use of available regex features.
 
-Try to create your regular expressions in a way that they match as few words as possible. E.g. using `hey[^\\s]*`
-instead of `hey.\*`, as the later one might match the whole message whereas the first one only matches a single word.
+The name of a regex in this case is a human readable description. It can help you remember what a regex is used for, and it is the title of the corresponding pattern feature. It does not have to match any intent or entity name. A regex for greeting might look like this:
```

Comment:
I'd say example of regex for such a general intent like greet is a bit dangerous. We actually don't want people to use regexes to identify intents like greet

Reply:
actually regex that matches one word is pointless, because it creates exactly the same feature as word level CountVectorsFeaturizer